EventListener/ReportSubscriber.php

```diff
@@ -0,0 +1,268 @@
+<?php
+
+namespace MauticPlugin\LeuchtfeuerCompanySegmentsBundle\EventListener;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\Expression\CompositeExpression;
+use Mautic\CoreBundle\Helper\InputHelper;
+use Mautic\LeadBundle\Model\CompanyReportData;
+use Mautic\ReportBundle\Event\ReportBuilderEvent;
+use Mautic\ReportBundle\Event\ReportGeneratorEvent;
+use Mautic\ReportBundle\ReportEvents;
+use MauticPlugin\LeuchtfeuerCompanySegmentsBundle\Entity\CompanySegmentRepository;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class ReportSubscriber implements EventSubscriberInterface
+{
+    public const CONTEXT_COMPANY_SEGMENTS         = 'company_segments';
+    public const COMPANY_TABLE                    = 'companies';
+    public const COMPANIES_PREFIX                 = 'comp';
+    public const COMPANY_SEGMENTS_XREF_PREFIX     = 'csx';
+    public const COMPANY_SEGMENTS_XREF_TABLE      = 'company_segment_xref';
+
+    public function __construct(
+        private CompanyReportData $companyReportData,
+        private CompanySegmentRepository $companySegmentRepository,
+        private Connection $db,
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            ReportEvents::REPORT_ON_BUILD    => ['onReportBuilder', 0],
+            ReportEvents::REPORT_ON_GENERATE => ['onReportGenerate', 0],
+        ];
+    }
+
+    public function onReportBuilder(ReportBuilderEvent $event): void
+    {
+        if (!$event->checkContext([self::CONTEXT_COMPANY_SEGMENTS])) {
+            return;
+        }
+
+        $columns = $this->companyReportData->getCompanyData();
+        unset($columns['companies_lead.is_primary'], $columns['companies_lead.date_added']);
```

Comment:
Here you also need to check that the columns are ones you expect, otherwise when Mautic adds something to the LeadList, which is not in the CompanySegment, we will have non-working filter.